packages/lib/src/utils/Transactions.js

```diff
@@ -152,12 +156,48 @@ async function getNodeVersion () {
   return state.nodeInfo;
 }
 
-async function checkGasPrice(txParams) {
-  if (await isGanacheNode()) return;
+async function getNetwork() {
+  if (!state.network) {
+    state.network = await promisify(web3.version.getNetwork.bind(global.web3.version))();
+  }
+  return state.network;
+}
+
+async function getETHGasStationPrice() {
+    if (state.gasPrice) return state.gasPrice;
+
+    try {
+      const apiResponse = await axios.get(GAS_API_URL);
+      const gasPriceGwei = apiResponse.average / 10;
+      const gasPrice = gasPriceGwei * 1e9;
+
+      state.gasPrice = gasPrice;
+      return state.gasPrice;
+    } catch (err) {
+      throw new Error(`Could not query gas price API to determine reasonable gas price, please provide one.`)
+    }
+}
+
+async function fixGasPrice(txParams) {
+  const network = await getNetwork();
+
   const gasPrice = txParams.gasPrice || Contracts.artifactsDefaults().gasPrice;
-  if (TRUFFLE_DEFAULT_GAS_PRICE.eq(gasPrice) || !gasPrice) {
-    throw new Error(`Cowardly refusing to execute transaction with excessively high default gas price of 100 gwei. Consider explicitly setting a different gasPrice value in your truffle.js config file. You can check reasonable values for gas price in https://ethgasstation.info/.`);
+
+  if (TRUFFLE_DEFAULT_GAS_PRICE == gasPrice || !gasPrice) {
```

Comment:
Let's use bignumber equality functions for this, ie `TRUFFLE_DEFAULT_GAS_PRICE.eq(gasPrice)`